### PR TITLE
Expire  cookie  that was set as part of token endpoint during logout

### PR DESCRIPTION
**Pull request by sudharshanraja-db**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

This change is to expire cookie during logout by exposing logout endpoint  , feature that was added as part of #542  and implemented as part of PR #593

### CODE REVIEW

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java [hunk 1]
```diff
@@ -167,17 +168,37 @@ public HttpResponse grantToken(
...
 
     return HttpResponse.ofJson(responseHeaders.build(), response);
   }
 
+  @Get("/logout")
+  public HttpResponse logout(HttpRequest request) {
+    String authorizationCookie =
+        request.headers().cookies().stream()
+            .filter(c -> c.name().equals(AuthDecorator.UC_TOKEN_KEY))
+            .map(Cookie::name)
```
Comment (jamieknight-db) [inline server/src/main/java/io/unitycatalog/server/service/AuthService.java:188]: @creechy @xx-db should this be `Cookie::value` like the change we made in AuthDecorator?

Comment (xx-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:188]: yes, this needs to be fixed in both places

Comment (jamieknight-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:188]: @sudharshanraja-db for reference, the change we made is in my[ UI PR here](https://github.com/unitycatalog/unitycatalog/pull/666/files#).

Comment (jamieknight-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:188]: It also looks like there are a lot of old commits included. I got myself into a similar situation and had to sync my forked repo with main repo, then reset HEAD to last commit.

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java [hunk 2]
```diff
@@ -167,17 +168,37 @@ public HttpResponse grantToken(
...
                 ServerProperties.getInstance().getProperty("server.cookie-timeout", "P5D");
             Cookie cookie =
-                Cookie.secureBuilder(AuthDecorator.UC_TOKEN_KEY, accessToken)
-                    .path("/")
-                    .maxAge(Duration.parse(cookieTimeout).getSeconds())
-                    .build();
+                createCookie(
+                    AuthDecorator.UC_TOKEN_KEY,
+                    accessToken,
+                    "/",
```
Comment (jaceklaskowski) [inline server/src/main/java/io/unitycatalog/server/service/AuthService.java:174]: Can we give this a name?

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java [hunk 3]
```diff
@@ -167,17 +168,37 @@ public HttpResponse grantToken(
...
+                    Duration.parse(cookieTimeout).getSeconds());
```
Comment (jaceklaskowski) [inline server/src/main/java/io/unitycatalog/server/service/AuthService.java:175]: Can this be moved to be part of `createCookie`?

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java [hunk 4]
```diff
@@ -167,17 +168,37 @@ public HttpResponse grantToken(
...
   }
 
+  @Get("/logout")
+  public HttpResponse logout(HttpRequest request) {
+    String authorizationCookie =
+        request.headers().cookies().stream()
+            .filter(c -> c.name().equals(AuthDecorator.UC_TOKEN_KEY))
+            .map(Cookie::name)
+            .findFirst()
+            .orElse(null);
```
Comment (jaceklaskowski) [inline server/src/main/java/io/unitycatalog/server/service/AuthService.java:190]: Why is this needed since you're doing `if` afterwards?

What do you think of getting rid of `orElse(null)` and chaining further to avoid `if` (that should really be called when there's non-`null` value and hence when `orElse` would not be needed)? It makes a lot of sense, doesn't it?

**Review by jaceklaskowski — CHANGES_REQUESTED** [review]

**Comment by jaceklaskowski** [comment]
I noticed you're on `sudharshanraja-db:main` branch. Create a feature branch instead.

**Comment by jaceklaskowski** [comment]
And remember to sign off your work. It hit me so many times, and it takes quite some time to work out after a couple of commits.

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 2 above)
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:174]: hi @jaceklaskowski  couldnt get you so  which part of the code you want to give name? is it the path parameter?

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 1 above)
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java]: @jamieknight-db  yes i made the changes to use Cookie::value and also regarding reset to head noted going forward will keep this in mind

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 3 above)
Comment (xx-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:175]: +1, the server property seems to be used only in createCookie.

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 2 above)
Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:174]: +1 what are you suggesting to "name"?

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 3 above)
Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:175]: +1

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 4 above)
Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:190]: Yeah, seems reasonable to just chain and either generate a `ResponseHeaders` with a new cookie, or an empty `ResponseHeaders`. Something like 
```
     request.headers().cookies().stream()
            .filter(c -> c.name().equals(AuthDecorator.UC_TOKEN_KEY))
            .findFirst()
            .map(ResponseHeaders.of(createCookie(...))
            .orElse(ResponseHeaders.of());
```

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 3 above)
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:175]: main idea behind put this in a method  as it is used  during login where we  set and other during logout where we  expire , the difference is during logout we need to set age to 0  and api accepts only Long  where as during login we accept duration in days .. so we need to convert it to long and then call the  create cookie method

**Comment by sudharshanraja-db** [comment]
> I noticed you're on `sudharshanraja-db:main` branch. Create a feature branch instead.

noted Thanks for suggestion  going forward for future PR will make sure this is followed

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 3 above)
Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:175]: Why not put the `Duration.parse(...).getSeconds()` part into `createCookie` so that you just pass in the duration string into `createCookie`? And then when you want to set age to zero, there should be a similar Duration syntax for that? `PT0S` maybe?

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java [hunk 5]
```diff
@@ -167,17 +168,35 @@ public HttpResponse grantToken(
...
         });
 
     return HttpResponse.ofJson(responseHeaders.build(), response);
   }
 
+  @Get("/logout")
+  public HttpResponse logout(HttpRequest request) {
+    return request.headers().cookies().stream()
+        .filter(c -> c.name().equals(AuthDecorator.UC_TOKEN_KEY))
+        .map(Cookie::value)
```
Comment (creechy) [inline server/src/main/java/io/unitycatalog/server/service/AuthService.java:187]: You don't need this anymore.

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 3 above)
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java]: ahh got it , didnt strike that 0 can be set in the format :)

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java (hunk 5 above)
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java]: Updated and also changed  logout to post

**Comment by sudharshanraja-db** [comment]
> @sudharshanraja-db looks like code formatting is failing, once that is fixed, this can be merged.

@jamieknight-db  there was an unused import fixed now  , didnt notice the failure email for the same
